### DISCUSSION TITLE
Error in testOwnerIsMsgSender() - i_owner not found/visible

### DISCUSSION THREAD

**Discussion by khandelwalmoksh** [Cyfrin/foundry-full-course-cu#2712] [Q&A] [answered]
my code(that too copied from written lesson) ``` function testOwnerIsMsgSender() public {
        assertEq(fundMe.i_owner(), address(this));
    } ```
    
![Screenshot (17)](https://github.com/user-attachments/assets/1c66a59d-0c0e-4ece-b031-0fe04321a78d)

**Comment by EngrPips** [comment]
Hello @khandelwalmoksh, check if the `i_owner` variable in your `FundMe` contract is public

**Reply by khandelwalmoksh** [reply]
constructor() {
        i_owner = msg.sender;
    }
No. But, constructor comes built in with public as specified by Patrick in lecture

**Reply by EngrPips** [reply]
I am not sure I understand what you mean but make the variable public and let's see if that resolve the error.

**Reply by khandelwalmoksh** [reply]
made Constructor public
error: Warning (2462)
made Modifier public
Error (2314)
![Screenshot (21)](https://github.com/user-attachments/assets/79afc036-1336-4904-9cc0-c9f03d87f5d0)

**Reply by EngrPips** [reply, marked as answer]
I said make the `i_owner` public, not the constructor.

**Reply by khandelwalmoksh** [reply]
I changed this ```address private immutable i_owner; ```
Success with a warninng.  tysm

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.
